pyanaconda/modules/payload/payloads/live_os/live_os.py

```diff
@@ -95,50 +95,54 @@ def _check_source_availability(self, message):
         if not self._image_source:
             raise SourceSetupError(message)
 
-    @property
-    def space_required(self):
-        """Get space required for the source image.
-
-        TODO: Add missing check if source is ready. Until then you shouldn't call this when
-        source is not ready.
-
-        TODO: This is not that fast as I thought (a few seconds). Caching or task?
+    @staticmethod
+    def _get_required_space():
+        # TODO: This is not that fast as I thought (a few seconds). Caching or solved in task?
+        size = get_dir_size("/") * 1024
 
-        :return: required size in bytes
-        :rtype: int
-        """
-        return get_dir_size("/") * 1024
+        # we don't know the size -- this should not happen
+        if size == 0:
+            log.debug("Space required is not known. This should not happen!")
+            return None
+        else:
+            return size
 
     def set_up_sources_with_task(self):
-        """Set up installation source."""
+        """Set up installation sources."""
         self._check_source_availability("Set up source failed - source is not set!")
 
-        return SetUpSourcesTask(self._sources)
+        task = SetUpSourcesTask(self._sources)
+        task.succeeded_signal.connect(lambda: self.set_required_space(self._get_required_space()))
+
+        return task
 
     def tear_down_sources_with_task(self):
         """Tear down installation sources."""
         self._check_source_availability("Tear down source failed - source is not set!")
 
-        return TearDownSourcesTask(self._sources)
+        task = TearDownSourcesTask(self._sources)
+        task.stopped_signal.connect(lambda: self.set_required_space(0))
+
+        return task
 
-    def pre_install_with_task(self):
-        """Prepare intallation task."""
+    def pre_install_with_tasks(self):
```

Comment:
pre -> prepare? To me it is really confusing. Why is one word shortened and the other not? I'd expect prepare_install vs pre_inst but here it's a bit of both. Or is it accepted terminology to call things "pre" in Anaconda?

Reply:
It's more historical reason. We are changing most of these things. Good point about this one, it should be changed.